File: plugins/plugin-elizacloud/rust/src/models/text.rs
#![allow(missing_docs)]

use crate::error::Result;
use crate::providers::ElizaCloudClient;
use crate::types::{ElizaCloudConfig, TextGenerationParams};

pub async fn handle_text_small(
    config: ElizaCloudConfig,
    params: TextGenerationParams,
) -> Result<String> {
    let client = ElizaCloudClient::new(config)?;
    client.generate_text_small(params).await
}

pub async fn handle_text_large(
    config: ElizaCloudConfig,
    params: TextGenerationParams,
) -> Result<String> {
    let client = ElizaCloudClient::new(config)?;
    client.generate_text_large(params).await
}

File: plugins/plugin-elizacloud/python/elizaos_plugin_elizacloud/models/text.py
from elizaos_plugin_elizacloud.providers import ElizaCloudClient
from elizaos_plugin_elizacloud.types import ElizaCloudConfig, TextGenerationParams


async def handle_text_small(
    config: ElizaCloudConfig,
    params: TextGenerationParams,
) -> str:
    async with ElizaCloudClient(config) as client:
        return await client.generate_text(params, model_size="small")


async def handle_text_large(
    config: ElizaCloudConfig,
    params: TextGenerationParams,
) -> str:
    async with ElizaCloudClient(config) as client:
        return await client.generate_text(params, model_size="large")
